src/Workspaces/Core/Portable/Extensions/ControlFlowRegionExtensions.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.FlowAnalysis
+{
+    internal static partial class ControlFlowRegionExtensions
+    {
+        internal static bool ContainsBlock(this ControlFlowRegion region, int destinationOrdinal)
+        {
+            return region.FirstBlockOrdinal <= destinationOrdinal && region.LastBlockOrdinal >= destinationOrdinal;
```

Comment:
can you help explain this?

Reply:
ControlFlowRegion is basically a range of contiguous basic blocks. There are different region kinds, such as try, catch, finally, etc. and this helper is checking if given basic block belongs to this region.

---
In reply to: [230606584](https://github.com/dotnet/roslyn/pull/30777#discussion_r230606584) [](ancestors = 230606584)